src/serialization/string_utils.cpp

```diff
@@ -822,47 +823,100 @@ std::vector<std::string> quoted_split(const std::string& val, char c, int flags,
 	return res;
 }
 
+namespace
+{
+/**
+ * Internal common code for parse_range and parse_range_real.
+ *
+ * If str contains two elements and a separator such as "a..b" or "a-b", returns a and b.
+ * Otherwise, returns the original string and std::nullopt.
+ */
+std::pair<std::string, std::optional<std::string>> parse_range_internal_separator(const std::string& str)
+{
+	// If adding additional options to this list, ensure that "-" (if present) is last. Otherwise a
+	// range such as "-2..-1" might be incorrectly split as "-2..", "1".
+	static const auto possible_separators = std::vector<std::string>{"..", "-"};
```

Comment:
Probably not a "now" thing but there _might_ be an argument for adding `~` to this list, since it's used in the animation syntax.

Reply:
Based on comments, I've removed `..` too, so it's just looking for `-`.